### PR TITLE
[TaskManager] Cleanup TODOs

### PR DESCRIPTION
**Pull request by scottn12**
## Description

This PR cleans up miscellaneous TODO items for TaskManager, such as the removal of debugging events/functions.

## Breaking Changes

This PR removes `TaskManager._getTaskQueues()`. Although this is technically a breaking change, this function was always intended to be used for debugging purposes. Additionally, TaskManager is currently an experimental package.

## Reviewer Guidance

[AB#2422](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2422)

### CODE REVIEW

File: packages/dds/task-manager/src/taskManager.ts [hunk 1]
```diff
@@ -647,7 +647,6 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         content.forEach(([taskId, clientIdQueue]) => {
             this.taskQueues.set(taskId, clientIdQueue);
         });
-        this.scrubClientsNotInQuorum();
```
Comment (ChumpChief) [inline packages/dds/task-manager/src/taskManager.ts:650]: This is exciting if we don't need it anymore!  Did you verify if this situation no longer occurs (e.g. with your fuzz test)?

**Review by ChumpChief — APPROVED** [review]
Looks good!  Might be interesting to use your fuzz tests to see how optimistic we can be about the queue state.

File: packages/dds/task-manager/src/taskManager.ts (hunk 1 above)
Comment (scottn12) [reply packages/dds/task-manager/src/taskManager.ts:650]: Yes! 😄

File: packages/dds/task-manager/src/taskManager.ts [hunk 2]
```diff
@@ -633,8 +626,15 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
...
         }
-        // TODO filter out tasks with no clients, some are still getting in.
-        const content = [...this.taskQueues.entries()];
+        const filteredMap = new Map<string, string[]>();
+        this.taskQueues.forEach((queue: string[], taskId: string) => {
+            if (queue === undefined || queue.length === 0) {
+                // Filter out tasks without any clients in the queue.
+                return;
+            }
+            filteredMap.set(taskId, queue);
```
[the remark this answers is not in the document]
Comment (scottn12) [reply packages/dds/task-manager/src/taskManager.ts:635]: Good call. Just tested and I think we can just remove this check all together 😄

**Pull request merged**